[CODE_OF_CONDUCT] Clause against leaked proprietary code (#5700)
reactos/reactos: CODE_OF_CONDUCT.md

Be respectful toward other members of the community. Don't initiate or engage in discussions that are designed to insult, inflame, attack, or incite hate/discrimination/bullying against others. Don't feed the trolls. If we detect any kind of behaviour we consider unacceptable, we reserve the right to ban, block, or remove the involved members from any or all of our communities.

> Never ever attempt to contribute leaked code from Microsoft Windows or any other proprietary software.

Reviewer: ```suggestion
Never ever attempt to contribute or even look at leaked code from Microsoft Windows or any other proprietary software.
```

> Never ever attempt to contribute or even look at leaked code from Microsoft Windows or any other proprietary software. Looking at but not copying proprietary code with expired copyright may or may not be acceptable depending on the specific artifact, and will be resolved on case-by-case basis.

Reviewer 2: ```suggestion
Never ever contribute or look at leaked code from Microsoft Windows or any other proprietary software. Looking at code protected by copyright is not acceptable. If you have seen Microsoft Windows source code, you are not allowed to contribute to this project.
```

Author: Prohibition of looking at out-of-copyright proprietary code except for case-by-case exceptions is intended, not a mistake.

Author: It is to prevent mass copying of code from a leak of, say, Windows happens to go out of copyright before it is officially open sourced.

Reviewer 3: > Looking at code protected by copyright is not acceptable.

This does not mean what you think it does. GPL code is protected why copyright -- in fact, that's the only reason the GPL works.

Reviewer 2: Fair criticism.

Reviewer 4: I would remove the 'attempt' from this sentence, because if you want to be anal you can say 'I did not attempt to do it, I did do it.'

Reviewer 2: I think the phrase I was looking for was 'prohibited by copyright' rather than 'protected by copyright'

> 

Reviewer 5: I'm not keen with the way you are phrasing your last sentence. `Looking at but not copying proprietary code with expired copyright may or may not` gives a **"you can do it if you try hard enough"** kind of vibe.

Whether a copyright expires or not is not the matter here but whether the licensed code explicitly allows such code to be looked at by the public and certainly in most cases a proprietary license won't allow that, regardless of whatever the case. Unless I'm misunderstanding something and the `copyright` in question you mean the license as well.

I would just keep the first sentence as is but if you want to be pedantic you may include `"Only the inclusion of open source third-party code by Microsoft is allowed`.

Author: @GeoB99 Whether copyright expires or not is indeed a matter, as the copyright law is the primary reason we do not want people to look at the source code of Windows. This clause extends this restriction to the work that has not been explicitly open sourced before the copyright expired, which should protect against paying public domain situations, i.e. a type of public domain where the royalty is paid to the state's copyright office instead, whereas free-for-all public domain is almost always safe to copy, and which kind of public domain is applicable depends on the jurisdiction of the original work's initial author. Algeria is the only country that I know that implements widespread paying public domain, i.e. requires royalty payment to the state after copyright expiration.

Reviewer 5: My point is that your current wording could confuse some people, because as it currently could be interpreted is that one would have access to the proprietary Windows source and look at it once, imagine, its copyright expires.

Access to the proprietary leaked Windows source is extremely prohibited, we should stay away from it and be firm on that. `may or may not be acceptable depending on the specific artifact` kind of relaxes this rule but what do I know, I'm not a native English speaker.

Author: >  one would have access to the proprietary Windows source and look at it once, imagine, its copyright expires.

The thing is that the copyright law is the reason we are not allowed to look at leaked Microsoft Windows source code in the first place. Once the copyright of a particular revision expires, it will indeed become a public domain piece of code (trade secrets, unlike copyrights, must be protected by the parties themselves, hence a leaked trade secret is no longer a secret for legal purposes). However, I have a proposed revision that forbids copying from Microsoft Windows, even well after its copyright expires:

```suggestion
Never ever attempt or act to contribute or even look at leaked code from Microsoft Windows or any other proprietary software. Looking at but not copying proprietary code with expired copyright may or may not be acceptable depending on the specific artifact, and will be resolved on case-by-case basis. However, code originally developed as a component of Microsoft Windows release is not acceptable, even if officially open sourced by Microsoft, even after copyright expiration. Pieces of code originally developed as a component of Microsoft Windows cannot be reused even if officially open sourced, even after the copyright of those code expires. Pieces of libraries open sourced by Microsoft which are also intended to be used in third party applications can however be reused. For those pieces of code, we require open sourcing to have happened during the copyright term under the copyright holder's own initiative.
```

Reviewer 5: I'm fine with that.

Reviewer 3: Why are we bothering with this whole "expired copyright" phrasing? That won't be a concern for another 30 years... if the project still exists then.

Author: @ThFabba Because Microsoft Windows is not the only such proprietary software that might have its copyright expired in our lifetimes.

Author: @ThFabba The case-by-case exception is intended for cases where the code is contributed with a specific formal permission of the original copyright owner under the condition of contribution under GPLv2 (i.e. GPLv2 if contributed to ReactOS first and acquired from it with proper attribution to ReactOS, ARR otherwise (quite similar to how some open access academic journals work)).

Author: @ThFabba Added a companion "exclusive licence with option to relicense to copyright" exception.
